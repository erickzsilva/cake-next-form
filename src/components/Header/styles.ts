import styled from 'styled-components'

export const Wrapper = styled.section`
  display: flex;
  max-width: 420px;
  margin: 0 auto;
  padding: 40px 10px 4px;
  flex-direction: column;
  width: 100%;
`
export const Title = styled.h1`
  font-size: 20px;
  font-weight: 500;
  margin-bottom: 15px;
}
`
export const SubTitle = styled.h2`
  font-size: 11px;
  font-weight: 400;
  margin-bottom: 10px;
  max-width: 335px;
`
