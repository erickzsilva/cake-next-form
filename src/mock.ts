import { CakeResponse } from './sharedTypes/types'

export const mock: CakeResponse = {
  title: 'Cake Order Form',
  description: `Order your freshly baked cakes using only the finest quality natural ingredientes.`,
  textChooseYourCake:
    'Please choose your favorite cake from among the following',
  subTitle: 'Order Information',
  cakes: [
    {
      value: 'Red Fruit Cake',
      name: 'redFruitCake',
      alt: 'Red Fruit Cake',
      srcImage: 'img/redFruitCake.jpg'
    },
    {
      value: 'Strawberry Sponge Cake',
      name: 'strawberrySpongeCake',
      alt: 'Strawberry Sponge Cake',
      srcImage: 'img/strawberrySpongeCake.jpg'
    },
    {
      value: 'Hazelnut Pie',
      name: 'hazelnutPie',
      alt: 'Hazelnut Pie',
      srcImage: 'img/hazelnutPie.jpg'
    },
    {
      value: 'MM Cake',
      name: 'mmCake',
      alt: 'MM Cake',
      srcImage: 'img/mmCake.jpg'
    }
  ],
  country: ['Argentina', 'Brasil', 'Fran√ßa', 'Mexico', 'Paraguai', 'Portugal']
}

export const mockHead = {
  title: 'Cake Order Form',
  description: `Order your freshly baked cakes using only the finest quality natural ingredientes.`
}
