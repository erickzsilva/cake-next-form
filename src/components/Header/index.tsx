'use client'
import * as S from './styles'

type HeaderPros = {
  title: string
  description: string
}

const Header = ({ title, description }: HeaderPros) => {
  return (
    <S.Wrapper>
      <S.Title title="title">{title}</S.Title>
      <S.SubTitle>{description}</S.SubTitle>
    </S.Wrapper>
  )
}

export default Header
