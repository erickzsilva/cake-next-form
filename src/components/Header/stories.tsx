import { Meta, StoryObj } from '@storybook/react'
import Header from '.'

export default {
  title: 'Header',
  component: Header
} as Meta

export const Default: StoryObj = {
  args: {
    title: 'Cake Order Form',
    description: `Order your freshly baked cakes using only the finest quality natural ingredientes.`
  }
}
