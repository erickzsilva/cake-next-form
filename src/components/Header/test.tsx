import { render, screen } from '@testing-library/react'

import { mockHead } from './../../mock'
import Header from '.'

describe('<Header />', () => {
  it('should render title and subtitle', () => {
    const { container } = render(
      <Header title={mockHead.title} description={mockHead.description} />
    )
    expect(
      screen.getByRole('heading', {
        name: /Cake Order Form/i
      })
    ).toBeInTheDocument()

    expect(
      screen.getByRole('heading', {
        name: /Order your freshly baked cakes using only the finest quality natural ingredientes./i
      })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
