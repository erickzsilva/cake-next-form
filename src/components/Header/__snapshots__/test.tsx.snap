// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render title and subtitle 1`] = `
.c0 {
  display: flex;
  max-width: 420px;
  margin: 0 auto;
  padding: 40px 10px 4px;
  flex-direction: column;
  width: 100%;
}

.c1 {
  font-size: 20px;
  font-weight: 500;
  margin-bottom: 15px;
}

.c2 {
  font-size: 11px;
  font-weight: 400;
  margin-bottom: 10px;
  max-width: 335px;
}

<section
  class="c0"
>
  <h1
    class="c1"
    title="title"
  >
    Cake Order Form
  </h1>
  <h2
    class="c2"
  >
    Order your freshly baked cakes using only the finest quality natural ingredientes.
  </h2>
</section>
`;
